WordPressKit/HTTPClient.swift

```diff
@@ -9,37 +9,92 @@ struct HTTPAPIResponse<Body> {
 
 extension URLSession {
 
+    /// Send a HTTP request and return its response as a `WordPressAPIResult` instance.
+    ///
+    /// ## Progress Tracking and Cancellation
+    ///
+    /// You can track the HTTP request's overall progress by passing a `Progress` instance to the `fulfillingProgress`
+    /// parameter, which must satisify following requirements:
+    /// - `totalUnitCount` must not be zero.
+    /// - `completedUnitCount` must be zero.
+    /// - It's used exclusivity for tracking the HTTP request overal progress: No children in its progress tree.
+    /// - `cancellationHandler` must be nil. You can call `fulfillingProgress.cancel()` to cancel the ongoing HTTP request.
```

Comment:
What do you think about encoding these constraints in either a dedicated type that wraps `Progress` or a subclass/pre-configured instance?

#### Type wrapping `Progress`

More coding and less flexibility for advanced users, but no chance of misuse since we'd be controlling everything from here

#### Subclass / Preconfigured instance

Could be misuses, but by letting the method accept a `Progress` type, we'd give power users more flexibility

Reply:
Update: I saw the `assert` below. They do a good job at ensuring runtime safety. Still, from a "how many things does a user need to know?" perspective, I think leveraging the type system somehow would be beneficial.